selene_sdk/sequences/genome.py

```diff
@@ -14,6 +14,60 @@
 from .sequence import sequence_to_encoding
 from .sequence import encoding_to_sequence
 
+def _check_coords(len_chrs,
+                  chrom,
+                  start,
+                  end,
+                  blacklist_tabix=None):
+    """
+    Check if the input coordinates are valid.
+
+    Parameters
+    ----------
+    len_chrs : dict
+        A dictionary mapping chromosome names to lengths.
+    chrom : str
+        The name of the chromosomes, e.g. "chr1".
+    start : int
+        The 0-based start coordinate of the sequence.
+    end : int
+        One past the last coordinate of the sequence.
+    blacklist_tabix : tabix.open or None, optional
+        Default is `None`. Tabix file handle if a file of blacklisted regions
+        is available.
+        
+    Returns
+    -------
+    bool
+        True if the coordinates are valid (`start` and `end` are within 
+        chromosome boundaries and not overlaping with blacklist regions
+        (if specified). Otherwise, return False.
+
+    Raises
+    ------
+    ValueError
+        If the input char to `strand` is not one of the specified
+        choices.
+    """
+    if chrom not in len_chrs:
+        return False
+
+    if start >= len_chrs[chrom]:
+        return False
+
+    if start >= end:
+        return False
+
+    if blacklist_tabix is not None:
```

Comment:
maybe turn this blacklist region check into a helper function, and then just write
```
return chrom in len_chrs and \
           start < len_chrs[chrom] and \
           start < end and \
           _not_blacklist_region(chrom, start, end)
```

Reply:
do we need to check for start >= 0 here too btw? not sure